R/call_derivation.R

```diff
@@ -94,23 +94,73 @@ call_derivation <- function(dataset = NULL, derivation, variable_params, ...) {
 
 #' Create a Set of Parameters
 #'
-#' Create a set of variable parameters to be used in an iteration of [`call_derivation()`]
+#' Create a set of variable parameters/function arguments to be used in [`call_derivation()`].
 #'
 #' @param ... One or more named arguments
 #'
-#' @author Thomas Neitmann
+#' @author Thomas Neitmann, Tracey Wang
 #'
 #' @return An object of class `params`
 #'
 #' @keywords source_specifications
 #'
 #' @export
 #'
+#' @seealso [call_derivation()]
+#'
 #' @examples
-#' params(
-#'   fns = list(VSSTRESN ~ mean(., na.rm = TRUE)),
-#'   set_values_to = vars(DTYPE = "AVERAGE")
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(admiral.test)
+#' data(ae)
+#' data(adsl)
+#'
+#' adae <-
+#'   select(ae[sample(1:nrow(ae), 1000),], USUBJID, AESTDTC, AEENDTC) %>%
+#'   derive_vars_merged(
+#'     dataset_add = adsl,
+#'     new_vars = vars(TRTSDT, TRTEDT),
+#'     by_vars = vars(USUBJID)
+#'   )
+#'
+#' ## In order to derive both `ASTDT` and `AENDT` in `ADAE`, one can use `derive_vars_dt()`
+#' adae %>%
+#'   derive_vars_dt(
+#'     new_vars_prefix = "AST",
+#'     dtc = AESTDTC,
+#'     date_imputation = "first",
+#'     min_dates = vars(TRTSDT),
+#'     max_dates = vars(TRTEDT)
+#'   ) %>%
+#'   derive_vars_dt(
+#'     new_vars_prefix = "AEN",
+#'     dtc = AEENDTC,
+#'     date_imputation = "last",
+#'     min_dates = vars(TRTSDT),
+#'     max_dates = vars(TRTEDT)
+#'   )
+#'
+#' ## As shown above, the same values were assigned to arguments `min_dates` and `max_dates`,
+#' ## whereas arguments `new_vars_prefix`, `dtc`, and `date_imputation` were assigned
+#' ## with different values to derive variables `ASTDT` and `AENDT`.
+#'
+#' ## While `derive_vars_dt()` can only add one variable at a time, using `call_derivation()`
+#' ## one can add multiple variables in one go. `params()` is used to create a set of arguments
+#' ## that are different from one another for each variable to be derived and form a `list` of
+#' ## arguments for `variable_params` in `call_derivation()`. The function arguments that stay the
+#' ## same among variables to be derived are specified outside `variable_params`.
```

Comment:
I would maybe re-phrase?
"While `derive_vars_dt()` can only add one variable at a time, using `call_derivation()` one can add multiple variables in one go. 
The function arguments which are different from a variable to another (e.g.  `new_vars_prefix`, `dtc`, and `date_imputation`) are specified as a `list` of `params()` in the `variable_params` argument of `call_derivation()`.
All other arguments which are common to all variables are specified outside of `variable_params` (i.e. in `...`).